napari_skimage_regionprops/_regionprops.py

```diff
@@ -184,3 +196,371 @@ def ellipsoid_axis_lengths(table):
+        value if it overcomes the 'overlap_threshold'. Otherwise, it returns 0.
+        In case of draws, it returns the first occurence. This function follows
+        the standards of skimage.regionprops 'extra_properties'.
+         
+        Parameters
+        ----------
+        regionmask : (M, N[,P]) ndarray
+            Label image (probe channel). Labels to be used as a mask.
+        intensity_image : (M, N[,P]) ndarray
+            Label image (reference channel). Labels to be measured using probe
+            channel as a mask.
+        
+        Returns
+        -------
+        value : int
+            Most frequent label number under regionmask, except 0, that
+            overcomes threshold. Otherwise, it returns 0.
+        """
+        if overlap_threshold == 0:
+            return 0
+        values, counts = np.unique(np.sort(intensity_image[regionmask]),
+                                   return_counts=True)
+        # Probabilities of belonging to a certain label or bg
+        probs = counts/np.sum(counts)
+        # If there is underlying bg, take it out
+        if values[0] == 0:
+            values = np.delete(values, 0)
+            probs = np.delete(probs, 0)
+
+        # if any label overlap probability is bigger than overlap_threshold
+        if (probs >= overlap_threshold).any():
+            # find label with highest overlap
+            # if equal frequency, return first occurence
+            index_max_overlap_prob = np.argmax(probs)
+            value = values[index_max_overlap_prob]
+        else:  # otherwise, highest allowed overlap is considered to be
+            # with background, i.e., object does not "belong" to any
+            #  other label and gets 0
+            value = 0
+        return value
+
+    # Measure properties of reference channel
+    if intensity_image is not None:
+        # If single intensity image was given, then use it
+        if intensity_image.shape[0]==1:
+            image = intensity_image[0]
+        # Otherwise use the corresponding intensity image
+        else:
+            image = intensity_image[ref_channel]
+    else:
+        image = None
+    ref_channel_props = regionprops_table(image=image,
+                                          labels=labels_array[ref_channel],
+                                          **kwargs)
+    # Add background measurement (basically NaNs)
+    ref_props_table = pd.DataFrame(ref_channel_props) # Convert to table
+    ref_props_table.loc[-1] = np.nan # Add first line with NaN
+    ref_props_table.index = ref_props_table.index + 1 # Increment indices
+    ref_props_table = ref_props_table.sort_index() # Sort rows by index
+    ref_props_table.iloc[0,0] = 0 # Add '0' to first cell (bg label is 0)
+    # Turn it back into dictionary
+    ref_channel_props = {name: ref_props_table[name].values for name in ref_props_table.columns}
+
+    for i in range(n_channels):
+        if i != ref_channel:
```

Comment:
See comment above. If the reference is an independent variable, we do not have to write special treatment code here.

Reply:
Agreed, this was removed.